ttnn/cpp/ttnn/operations/moreh/moreh_fold/device/fold_program_factory_rm.cpp

```diff
@@ -0,0 +1,218 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include <vector>
+
+#include "fold_device_operation.hpp"
+#include "tt_metal/common/work_split.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/operations/moreh/moreh_helper_functions.hpp"
+
+namespace ttnn::operations::moreh::moreh_fold {
+
+using namespace tt;
+using namespace tt::tt_metal;
+
+MorehFoldOperation::ProgramFactory::cached_program_t MorehFoldOperation::ProgramFactory::create(
+    const operation_attributes_t& operation_attributes,
+    const tensor_args_t& tensor_args,
+    tensor_return_value_t& output_tensor) {
+    auto& input = tensor_args.input;
+    auto& output = output_tensor;
```

Comment:
I think you should use the output_tensor directly (Btw, change the name to output if you can)

Reply:
Resolved in latest commit